src/gpuarray/buffer.h

```diff
@@ -634,8 +637,16 @@ GPUARRAY_PUBLIC gpucontext *gpukernel_context(gpukernel *k);
  */
 #define GA_CTX_PROP_MAXLSIZE2 17
 
+/**
+ * Get the vector of collective ops for the context.
+ *
+ * Type: `const gpuarray_comm_ops *`
+ */
+#define GA_CTX_PROP_COMM_OPS  18
```

Comment:
I still cannot find what do you mean. Do you want to replace the "properties" macros with a enum defined inside the `gpucontext`?